lib/src/main.dart

```diff
@@ -1,14 +1,32 @@
 import 'dart:async';
 
+import 'package:clock/clock.dart';
 import 'package:connectivity_plus/connectivity_plus.dart';
+import 'package:flutter/scheduler.dart';
 import 'package:flutter/widgets.dart';
 import 'package:flutter_offline/src/utils.dart';
 import 'package:network_info_plus/network_info_plus.dart';
 
 const kOfflineDebounceDuration = Duration(seconds: 3);
+const kDefaultMaxRetries = 5;
+const kDefaultRetryCooldown = Duration(seconds: 2);
 
 typedef ValueWidgetBuilder<T> = Widget Function(BuildContext context, T value, Widget child);
 
+/// Controller for managing retry callbacks
+class RetryController {
+  RetryController({
+    this.onRetry,
+    this.onRetryError,
+  });
+
+  /// Callback function that gets executed when retry is triggered
+  final Future<void> Function()? onRetry;
+
+  /// Callback function that gets executed when retry fails
+  final void Function(Object error, StackTrace stackTrace)? onRetryError;
```

Comment:
I would prefer if these are methods the caller could implement `OfflineRetryController` and override